src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -191,6 +192,7 @@ public void execute(BuildResult buildResult) {
                 Optional<String> stageName = envVariable("STAGE_NAME", providers);
                 Optional<String> controllerUrl = envVariable(controllerUrlEnvVar, providers);
 
+                buildScan.value("CI provider", ciProvider);
```

Comment:
We typically don't have constants for custom value keys but it might make sense here because this one is repeated.

Reply:
For my taste, I would not add a constant here.